### PR TITLE
Java : Exception Handling

### PR DESCRIPTION
**Pull request by Ritabrata1080**
- [x] Added comments for describing functions
- [x] Updated file header
- [x] No new warning added by PR
- [x] Squashed all commits into 1 before raising PR

### CODE REVIEW

File: src/ExceptionHandling.java [hunk 1]
```diff
@@ -32,12 +32,44 @@ public void getLength(String s){
...
+            System.out.println(a[index]);
+        } catch (ArrayIndexOutOfBoundsException e){
+            System.out.println(e.getMessage());
+        }
+    }
 }
 
 public class ExceptionHandling {
     public static void main(String[] args) {
         PerformOperations obj = new PerformOperations();
```
Comment (avanura23) [inline src/ExceptionHandling.java:62]: I'd suggest to use meaningful names for object creation.

File: src/ExceptionHandling.java [hunk 2]
```diff
@@ -32,12 +32,44 @@ public void getLength(String s){
...
         obj.performDivision(100, 0);
         obj.getLength(null);
+        int[] arr = null;
+        int res = obj.getArrayLength(null);
+        System.out.println(res);
```
Comment (avanura23) [inline src/ExceptionHandling.java:67]: As of now println() statements are OK. 
I'd suggest to look into loggers and how they can be used.

File: src/ExceptionHandling.java [hunk 3]
```diff
@@ -32,12 +32,44 @@ public void getLength(String s){
...
+        int[] testArray = new int[3];
+        testArray[0] = 1;
+        testArray[1] = 2;
+        testArray[2] = 3;
```
Comment (avanura23) [inline src/ExceptionHandling.java:71]: Not sure why we are assigning array values in this fashion when the elements are sequential.

File: src/ExceptionHandling.java [hunk 4]
```diff
@@ -32,12 +32,44 @@ public void getLength(String s){
...
+        return 0;
+    }
+
+    // Making aware that this API could throw an exception. Possibly to be handled with try catch
+    public void getArrayIndexThrowException(int[] a, int index) throws ArrayIndexOutOfBoundsException {
+        System.out.println(a[index]);
+    }
+
+    //Handling the exception through try catch block
+    public void getArrayIndexHandleException(int[] a, int index) {
```
Comment (avanura23) [inline src/ExceptionHandling.java:51]: Is it possible to club the exception handling of this method and method on line 36 with a single implementation?

File: src/ExceptionHandling.java [hunk 5]
```diff
@@ -32,12 +32,44 @@ public void getLength(String s){
             System.out.println("Exception cause : " + np.getMessage());
         }
     }
+
+    public int getArrayLength(int[] a){
```
Comment (avanura23) [inline src/ExceptionHandling.java:36]: Check comment on line 51

File: src/ExceptionHandling.java (hunk 2 above)
Comment (Ritabrata1080) [reply src/ExceptionHandling.java:67]: Yes, we need to use loggers to maintain industry standard codes. But this is just for learning purpose and revise the exception handling topics.

File: src/ExceptionHandling.java (hunk 1 above)
Comment (Ritabrata1080) [reply src/ExceptionHandling.java:62]: Yes.

File: src/ExceptionHandling.java (hunk 4 above)
Comment (Ritabrata1080) [reply src/ExceptionHandling.java:51]: Previous function is showing nullpointer exception case whereas this function is showing ArrayIndexOutOfBounds exception. Moreover, it's showing how to handle exception with try catch and in contrast to this, the function at line number 46 is written, where we are throwing the exception in place of handling it with try-catch.

File: src/ExceptionHandling.java (hunk 3 above)
Comment (Ritabrata1080) [reply src/ExceptionHandling.java:71]: addressed. Initialized normally without explicitly defining every element.

File: src/ExceptionHandling.java (hunk 4 above)
Comment (avanura23) [reply src/ExceptionHandling.java:51]: Both the exceptions are of the type RuntimeException. Create a separate class which will handle the exception(s) which can be reused in the future.

**Review by AGNIVA029 — APPROVED** [review]

**Review by avanura23 — APPROVED** [review]

**Pull request merged**